Can't use Typescript types due to wrong packaging

### Describe the Bug

I'm trying to use form-js in an Angular project, but it seems that the files aren't correctly packaged. 

I'm seeing the following issues:

```
...
Error: ../node_modules/@bpmn-io/form-js-viewer/dist/types/Form.d.ts:127:31 - error TS2307: Cannot find module './types' or its corresponding type declarations.

127 export type Injector = import('./types').Injector;
                                  ~~~~~~~~~


Error: ../node_modules/@bpmn-io/form-js-viewer/dist/types/Form.d.ts:128:27 - error TS2307: Cannot find module './types' or its corresponding type declarations.

128 export type Data = import('./types').Data;
                              ~~~~~~~~~


Error: ../node_modules/@bpmn-io/form-js-viewer/dist/types/Form.d.ts:129:29 - error TS2307: Cannot find module './types' or its corresponding type declarations.

129 export type Errors = import('./types').Errors;
                                ~~~~~~~~~


Error: ../node_modules/@bpmn-io/form-js-viewer/dist/types/Form.d.ts:130:29 - error TS2307: Cannot find module './types' or its corresponding type declarations.

130 export type Schema = import('./types').Schema;
                                ~~~~~~~~~
...
```

Part of Form.d.ts
```
...
export type Injector = import('./types').Injector;
export type Data = import('./types').Data;
export type Errors = import('./types').Errors;
export type Schema = import('./types').Schema;
export type FormProperties = import('./types').FormProperties;
export type FormProperty = import('./types').FormProperty;
export type FormEvent = import('./types').FormEvent;
export type FormOptions = import('./types').FormOptions;
...
```

So it seems that the file `@bpmn-io/form-js-viewer/dist/types/Form.d.ts` is trying to read the file `types` in the current directory, but the actual `types.d.ts` file is still in the `src/` subfolder.


```
.
├── dist
│   ├── assets
│   │   ├── flatpickr
│   │   │   └── light.css
│   │   └── form-js.css
│   ├── index.cjs
│   ├── index.cjs.map
│   ├── index.es.js
│   ├── index.es.js.map
│   └── types
│       ├── core
│       │   └── ...
│       ├── import
│       │   └── ...
│       ├── render
│       │   └── ...
│       ├── src
│       │   └── types.d.ts
│       └── util
│       │   └── ...
│       ├── Form.d.ts
│       ├── index.d.ts
├── LICENSE
├── package.json
└── README.md
```

### Steps to Reproduce

<!-- Clear steps that allow us to reproduce the error. -->

1. npm install @bpmn-io/form-js-viewer
2. Add `import {Form} from '@bpmn-io/form-js-viewer';` on a typescript file.
2. Try to compile in an typescript project
3. It's complaining that the file `types` can't be found.

I think it's related to https://github.com/bpmn-io/form-js/pull/143.


### Expected Behavior

<!-- A clear and concise description of what you expected to happen. -->


### Environment

- Host: Node v16.17.1 / Angular 12
- OS: Ubuntu 22.04
- Library version: 0.11.0

# Minimal steps to reproduce


```ts
# index.ts
import { Form } from '@bpmn-io/form-js-viewer';
```


```
$ cd /tmp/
$ mkdir formjs-test/
$ cd formjs-test/
$ npm install @bpmn-io/form-js-viewer typescript
$ echo "import { Form } from '@bpmn-io/form-js-viewer';" > index.ts
$ cat index.ts
import { Form } from '@bpmn-io/form-js-viewer';
$ node_modules/.bin/tsc index.ts
node_modules/@bpmn-io/form-js-viewer/dist/types/Form.d.ts:127:31 - error TS2307: Cannot find module './types' or its corresponding type declarations.

127 export type Injector = import('./types').Injector;
                                  ~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/Form.d.ts:128:27 - error TS2307: Cannot find module './types' or its corresponding type declarations.

128 export type Data = import('./types').Data;
                              ~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/Form.d.ts:129:29 - error TS2307: Cannot find module './types' or its corresponding type declarations.

129 export type Errors = import('./types').Errors;
                                ~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/Form.d.ts:130:29 - error TS2307: Cannot find module './types' or its corresponding type declarations.

130 export type Schema = import('./types').Schema;
                                ~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/Form.d.ts:131:37 - error TS2307: Cannot find module './types' or its corresponding type declarations.

131 export type FormProperties = import('./types').FormProperties;
                                        ~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/Form.d.ts:132:35 - error TS2307: Cannot find module './types' or its corresponding type declarations.

132 export type FormProperty = import('./types').FormProperty;
                                      ~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/Form.d.ts:133:32 - error TS2307: Cannot find module './types' or its corresponding type declarations.

133 export type FormEvent = import('./types').FormEvent;
                                   ~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/Form.d.ts:134:34 - error TS2307: Cannot find module './types' or its corresponding type declarations.

134 export type FormOptions = import('./types').FormOptions;
                                     ~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/Form.d.ts:142:13 - error TS2456: Type alias 'OnEventWithPriority' circularly references itself.

142 export type OnEventWithPriority = OnEventWithPriority;
                ~~~~~~~~~~~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/Form.d.ts:143:13 - error TS2456: Type alias 'OnEventWithOutPriority' circularly references itself.

143 export type OnEventWithOutPriority = OnEventWithOutPriority;
                ~~~~~~~~~~~~~~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/core/ConditionChecker.d.ts:30:44 - error TS2307: Cannot find module '../types' or its corresponding type declarations.

30     check(condition: string, data?: import('../types').Data): boolean | null;
                                              ~~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/core/ConditionChecker.d.ts:49:48 - error TS2307: Cannot find module '../types' or its corresponding type declarations.

49     evaluate(expression: string, data?: import('../types').Data): any;
                                                  ~~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/index.d.ts:15:40 - error TS2307: Cannot find module './types' or its corresponding type declarations.

15 export type CreateFormOptions = import('./types').CreateFormOptions;
                                          ~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/render/components/icons/index.d.ts:2:14 - error TS2304: Cannot find name 'ButtonIcon'.

2     export { ButtonIcon as button };
               ~~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/render/components/icons/index.d.ts:3:14 - error TS2304: Cannot find name 'CheckboxIcon'.

3     export { CheckboxIcon as checkbox };
               ~~~~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/render/components/icons/index.d.ts:4:14 - error TS2304: Cannot find name 'ChecklistIcon'.

4     export { ChecklistIcon as checklist };
               ~~~~~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/render/components/icons/index.d.ts:5:14 - error TS2304: Cannot find name 'ColumnsIcon'.

5     export { ColumnsIcon as columns };
               ~~~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/render/components/icons/index.d.ts:6:14 - error TS2304: Cannot find name 'DatetimeIcon'.

6     export { DatetimeIcon as datetime };
               ~~~~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/render/components/icons/index.d.ts:7:14 - error TS2304: Cannot find name 'ImageIcon'.

7     export { ImageIcon as image };
               ~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/render/components/icons/index.d.ts:8:14 - error TS2304: Cannot find name 'NumberIcon'.

8     export { NumberIcon as number };
               ~~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/render/components/icons/index.d.ts:9:14 - error TS2304: Cannot find name 'RadioIcon'.

9     export { RadioIcon as radio };
               ~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/render/components/icons/index.d.ts:10:14 - error TS2304: Cannot find name 'SelectIcon'.

10     export { SelectIcon as select };
                ~~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/render/components/icons/index.d.ts:10:14 - error TS2552: Cannot find name 'SelectIcon'. Did you mean 'Selection'?

10     export { SelectIcon as select };
                ~~~~~~~~~~

  node_modules/typescript/lib/lib.dom.d.ts:13621:13
    13621 declare var Selection: {
                      ~~~~~~~~~
    'Selection' is declared here.

node_modules/@bpmn-io/form-js-viewer/dist/types/render/components/icons/index.d.ts:11:14 - error TS2304: Cannot find name 'TaglistIcon'.

11     export { TaglistIcon as taglist };
                ~~~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/render/components/icons/index.d.ts:12:14 - error TS2304: Cannot find name 'TextIcon'.

12     export { TextIcon as text };
                ~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/render/components/icons/index.d.ts:13:14 - error TS2304: Cannot find name 'TextfieldIcon'.

13     export { TextfieldIcon as textfield };
                ~~~~~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/render/components/icons/index.d.ts:14:14 - error TS2304: Cannot find name 'TextareaIcon'.

14     export { TextareaIcon as textarea };
                ~~~~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/render/components/icons/index.d.ts:15:14 - error TS2304: Cannot find name 'FormIcon'.

15     export { FormIcon as default };
                ~~~~~~~~


Found 28 errors in 4 files.

Errors  Files
    10  node_modules/@bpmn-io/form-js-viewer/dist/types/Form.d.ts:127
     2  node_modules/@bpmn-io/form-js-viewer/dist/types/core/ConditionChecker.d.ts:30
     1  node_modules/@bpmn-io/form-js-viewer/dist/types/index.d.ts:15
    15  node_modules/@bpmn-io/form-js-viewer/dist/types/render/components/icons/index.d.ts:2
```

I've just tried with 0.9, but I still see the circularly references there. The type file seems to be in the correct place.

```
$ npm install @bpmn-io/form-js-viewer@0.9 typescript
$ node_modules/.bin/tsc index.ts
node_modules/@bpmn-io/form-js-viewer/dist/types/Form.d.ts:138:13 - error TS2456: Type alias 'OnEventWithPriority' circularly references itself.

138 export type OnEventWithPriority = OnEventWithPriority;
                ~~~~~~~~~~~~~~~~~~~

node_modules/@bpmn-io/form-js-viewer/dist/types/Form.d.ts:139:13 - error TS2456: Type alias 'OnEventWithOutPriority' circularly references itself.

139 export type OnEventWithOutPriority = OnEventWithOutPriority;
                ~~~~~~~~~~~~~~~~~~~~~~


Found 2 errors in the same file, starting at: node_modules/@bpmn-io/form-js-viewer/dist/types/Form.d.ts:138
```

> I've just tried with 0.9, but I still see the circularly references there. The type file seems to be in the correct place.
> 
> ```
> $ npm install @bpmn-io/form-js-viewer@0.9 typescript
> $ node_modules/.bin/tsc index.ts
> node_modules/@bpmn-io/form-js-viewer/dist/types/Form.d.ts:138:13 - error TS2456: Type alias 'OnEventWithPriority' circularly references itself.
> 
> 138 export type OnEventWithPriority = OnEventWithPriority;
>                 ~~~~~~~~~~~~~~~~~~~
> 
> node_modules/@bpmn-io/form-js-viewer/dist/types/Form.d.ts:139:13 - error TS2456: Type alias 'OnEventWithOutPriority' circularly references itself.
> 
> 139 export type OnEventWithOutPriority = OnEventWithOutPriority;
>                 ~~~~~~~~~~~~~~~~~~~~~~
> 
> 
> Found 2 errors in the same file, starting at: node_modules/@bpmn-io/form-js-viewer/dist/types/Form.d.ts:138
> ```

you just need to replace those line with this it should fix it : 
export type OnEventWithPriority = (type: FormEvent, priority: number, handler: Function) => void;
export type  OnEventWithOutPriority = (type: FormEvent, handler: Function) => void;
export type OnEventType = OnEventWithPriority & OnEventWithOutPriority;

--- a/package.json
+++ b/package.json
@@ -101,7 +101,7 @@
     "sinon": "^15.0.0",
     "sinon-chai": "^3.7.0",
     "source-map-loader": "^4.0.0",
-    "typescript": "^4.3.4",
+    "typescript": "^4.9.5",
     "webpack": "^5.40.0"
   }
 }

--- a/packages/form-js-editor/package.json
+++ b/packages/form-js-editor/package.json
@@ -27,7 +27,7 @@
     "example:dev": "cd example && npm start",
     "lint": "run-s lint:*",
     "lint:eslint": "eslint .",
-    "generate-types": "tsc --allowJs --skipLibCheck --declaration --emitDeclarationOnly --outDir dist/types src/index.js && copyfiles src/**.d.ts dist/types",
+    "generate-types": "tsc --allowJs --skipLibCheck --declaration --emitDeclarationOnly --outDir dist/types src/index.js && copyfiles --flat src/**.d.ts dist/types",
     "test": "cross-env NODE_ENV=test karma start",
     "prepublishOnly": "npm run build"
   },

--- a/packages/form-js-editor/src/features/palette/components/Palette.js
+++ b/packages/form-js-editor/src/features/palette/components/Palette.js
@@ -110,7 +110,7 @@ export default function Palette(props) {
             <div class="fjs-palette-fields fjs-drag-container fjs-no-drop">
               {
                 entries.map(({ label, type }) => {
-                  const Icon = iconsByType[type];
+                  const Icon = iconsByType(type);
 
                   return (
                     <div

--- a/packages/form-js-editor/src/features/properties-panel/PropertiesPanelHeaderProvider.js
+++ b/packages/form-js-editor/src/features/properties-panel/PropertiesPanelHeaderProvider.js
@@ -48,7 +48,7 @@ export const PropertiesPanelHeaderProvider = {
       type
     } = field;
 
-    const Icon = iconsByType[type];
+    const Icon = iconsByType(type);
 
     if (Icon) {
       return () => <Icon width="36" height="36" viewBox="0 0 54 54" />;

--- a/packages/form-js-editor/src/render/components/FormEditor.js
+++ b/packages/form-js-editor/src/render/components/FormEditor.js
@@ -397,7 +397,7 @@ function CreatePreview(props) {
 
     const fieldType = clone.dataset.fieldType;
 
-    const Icon = iconsByType[ fieldType ];
+    const Icon = iconsByType(fieldType);
 
     const { label } = findPaletteEntry(fieldType);
 

--- a/packages/form-js-editor/src/render/components/editor-form-fields/EditorText.js
+++ b/packages/form-js-editor/src/render/components/editor-form-fields/EditorText.js
@@ -7,7 +7,7 @@ export default function EditorText(props) {
 
   const { type, text = '' } = props.field;
 
-  const Icon = iconsByType['text'];
+  const Icon = iconsByType('text');
 
   if (!text) {
     return <div class={ editorFormFieldClasses(type) }>

--- a/packages/form-js-integration/package.json
+++ b/packages/form-js-integration/package.json
@@ -3,6 +3,10 @@
   "version": "0.12.0",
   "description": "A set of form-js integration testing resources",
   "private": true,
+  "scripts": {
+    "test": "run-s test:types",
+    "test:types": "tsc src/application.ts --outDir dist"
+  },
   "repository": {
     "type": "git",
     "url": "git+https://github.com/bpmn-io/form-js.git",

--- a/packages/form-js-integration/src/application.ts
+++ b/packages/form-js-integration/src/application.ts
@@ -3,6 +3,7 @@ import {
   createFormEditor,
   Form,
   FormEditor,
+  FormPlayground,
   schemaVersion
 } from '@bpmn-io/form-js';
 
@@ -54,4 +55,10 @@ export async function createApp() {
 
   // eslint-disable-next-line
   const detachedFormEditor = new FormEditor();
+
+  // eslint-disable-next-line
+   const detachedPlayground = new FormPlayground({
+    schema: {},
+    data: {}
+  });
 }

--- a/packages/form-js-viewer/package.json
+++ b/packages/form-js-viewer/package.json
@@ -24,7 +24,7 @@
     "bundle": "rollup -c --failAfterWarnings --bundleConfigAsCjs",
     "bundle:watch": "rollup -c -w --bundleConfigAsCjs",
     "dev": "npm test -- --auto-watch --no-single-run",
-    "generate-types": "tsc --allowJs --skipLibCheck --declaration --emitDeclarationOnly --outDir dist/types src/index.js && copyfiles src/*.d.ts dist/types",
+    "generate-types": "tsc --allowJs --skipLibCheck --declaration --emitDeclarationOnly --outDir dist/types src/index.js && copyfiles --flat src/*.d.ts dist/types",
     "test": "karma start",
     "prepublishOnly": "npm run build"
   },

--- a/packages/form-js-viewer/src/render/components/icons/index.js
+++ b/packages/form-js-viewer/src/render/components/icons/index.js
@@ -13,19 +13,21 @@ import TextfieldIcon from './Textfield.svg';
 import TextareaIcon from './Textarea.svg';
 import ImageIcon from './Image.svg';
 
-export const iconsByType = {
-  button: ButtonIcon,
-  checkbox: CheckboxIcon,
-  checklist: ChecklistIcon,
-  columns: ColumnsIcon,
-  datetime: DatetimeIcon,
-  image: ImageIcon,
-  number: NumberIcon,
-  radio: RadioIcon,
-  select: SelectIcon,
-  taglist: TaglistIcon,
-  text: TextIcon,
-  textfield: TextfieldIcon,
-  textarea: TextareaIcon,
-  default: FormIcon
+export const iconsByType = (type) => {
+  return {
+    button: ButtonIcon,
+    checkbox: CheckboxIcon,
+    checklist: ChecklistIcon,
+    columns: ColumnsIcon,
+    datetime: DatetimeIcon,
+    image: ImageIcon,
+    number: NumberIcon,
+    radio: RadioIcon,
+    select: SelectIcon,
+    taglist: TaglistIcon,
+    text: TextIcon,
+    textfield: TextfieldIcon,
+    textarea: TextareaIcon,
+    default: FormIcon
+  }[type];
 };

--- a/packages/form-js/package.json
+++ b/packages/form-js/package.json
@@ -43,6 +43,7 @@
     "test:distro": "run-s test:distro:*",
     "test:distro:viewer": "cross-env VARIANT=form-viewer karma start test/config/karma.distro.js",
     "test:distro:editor": "cross-env VARIANT=form-editor karma start test/config/karma.distro.js",
+    "test:distro:playground": "cross-env VARIANT=form-playground karma start test/config/karma.distro.js",
     "prepublishOnly": "run-s build test:distro"
   },
   "license": "SEE LICENSE IN LICENSE",

